spec/services/attribute_asserter_spec.rb

```diff
@@ -688,21 +686,107 @@
         subject.build
       end
 
-      it 'includes ial' do
-        expect(user.asserted_attributes.keys).to include(:ial)
-      end
-
-      it 'creates a getter function for ial attribute' do
+      it 'asserts IAL1' do
         expected_ial = Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
-        expect(user.asserted_attributes[:ial][:getter].call(user)).to eq expected_ial
+        expect(get_asserted_attribute(user, :ial)).to eq expected_ial
       end
 
-      it 'includes proofed attributes' do
+      it 'does not include proofed attributes' do
         expect(user.asserted_attributes[:first_name]).to eq(nil)
         expect(user.asserted_attributes[:phone]).to eq(nil)
       end
     end
 
+    context 'when biometric IAL preferred is requested' do
+      let(:subject) do
+        described_class.new(
+          user: user,
+          name_id_format: name_id_format,
+          service_provider: service_provider,
+          authn_request: biometric_preferred_ial_authnrequest,
+          decrypted_pii: decrypted_pii,
+          user_session: user_session,
+        )
+      end
+
+      context 'when the user has been proofed with biometric comparison' do
+        let(:user) { biometric_verified_user }
+        before do
+          user.identities << identity
+          subject.build
+        end
+
+        it 'asserts IAL2 with biometric comparison' do
+          expected_ial = Saml::Idp::Constants::IAL2_BIO_REQUIRED_AUTHN_CONTEXT_CLASSREF
+          expect(get_asserted_attribute(user, :ial)).to eq expected_ial
+        end
+      end
+
+      context 'when the user has been proofed but not with biometric comparison' do
+        before do
+          user.identities << identity
+          subject.build
+        end
+
+        it 'asserts IAL2 (without biometric comparison)' do
+          expected_ial = Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
+          expect(get_asserted_attribute(user, :ial)).to eq expected_ial
+        end
+      end
+    end
+
+    context 'when IAL1 is requested' do
```

Comment:
hm there is already an IAL1 context. (line 365). can these tests go in there, so that all the IAL1 requested tests are together?